packages/blockchain/src/db/manager.ts

```diff
@@ -109,30 +102,27 @@ export class DBManager {
       throw new Error('Unknown blockId type')
     }
 
+    if (hash === undefined || number === undefined) return undefined
     const header = await this.getHeader(hash, number)
-    let body: BlockBodyBytes
-    try {
-      body = await this.getBody(hash, number)
-    } catch (error: any) {
-      if (error.code !== 'LEVEL_NOT_FOUND') {
-        throw error
-      }
-
+    const body = await this.getBody(hash, number)
+    if (body[0].length === 0 && body[1].length === 0) {
       // Do extra validations on the header since we are assuming empty transactions and uncles
-      if (
-        !equalsBytes(header.transactionsTrie, KECCAK256_RLP) ||
-        !equalsBytes(header.uncleHash, KECCAK256_RLP_ARRAY)
-      ) {
-        throw error
+      if (!equalsBytes(header.transactionsTrie, KECCAK256_RLP)) {
+        throw new Error('transactionsTrie root should be equal to hash of null')
+      }
+      if (!equalsBytes(header.uncleHash, KECCAK256_RLP_ARRAY)) {
+        throw new Error('uncle hash should be equal to hash of empty array')
       }
-      body = [[], []]
       // If this block had empty withdrawals push an empty array in body
       if (header.withdrawalsRoot !== undefined) {
         // Do extra validations for withdrawal before assuming empty withdrawals
-        if (!equalsBytes(header.withdrawalsRoot, KECCAK256_RLP)) {
-          throw error
+        if (
+          !equalsBytes(header.withdrawalsRoot, KECCAK256_RLP) &&
+          (body.length !== 3 || body[2]?.length === 0)
+        ) {
+          throw new Error('withdrawals root shoot be equal to hash of null when no withdrawals')
         }
-        body.push([])
+        if (body.length !== 3) body.push([])
```

Comment:
Not have the complete picture here, is the case at this point that body length might equal 3 sufficiently covered or alternatively not possible to reach due to previous code?